csharp/src/Ice/OutgoingRequestFrame.cs

```diff
@@ -36,41 +94,32 @@ public sealed class OutgoingRequestFrame : OutputStream
         /// proxy and the communicator's current context (if any).</param>
         public static OutgoingRequestFrame Empty(IObjectPrx proxy, string operation, bool idempotent,
                                                  IReadOnlyDictionary<string, string>? context = null)
-            => new OutgoingRequestFrame(proxy, operation, idempotent, context, ArraySegment<byte>.Empty);
+            => new OutgoingRequestFrame(proxy, operation, idempotent, true, context);
 
-        /// <summary>Creates a new outgoing request frame. This frame is incomplete and its payload needs to be
-        /// provided using StartParameters/EndParameters.</summary>
-        /// <param name="proxy">A proxy to the target Ice object. This method uses the communicator, identity, facet,
-        /// encoding and context of this proxy to create the request frame.</param>
-        /// <param name="operation">The operation to invoke on the target Ice object.</param>
-        /// <param name="idempotent">True when operation is idempotent, otherwise false.</param>
-        /// <param name="context">An optional explicit context. When non null, it overrides both the context of the
-        /// proxy and the communicator's current context (if any).</param>
-        public OutgoingRequestFrame(IObjectPrx proxy, string operation, bool idempotent,
-                                    IReadOnlyDictionary<string, string>? context = null)
-            : base(proxy.Communicator)
+        private OutgoingRequestFrame(IObjectPrx proxy, string operation, bool idempotent,
```

Comment:
I would move this constructor to the end of the class, remove the default for the context parameter, make writeEmptyEncaps as the last parameter and make the parameter explicit (as in `writeEmptyEncaps: false`) when calling it from other constructors.